contracts/MarginAccount.sol

```diff
@@ -0,0 +1,189 @@
+
+    /**
+     * @dev Account is a struct corresponding to a user that describes how many vaults that user has.
+     */
+    struct Account {
+        //@dev the owner whose vaults we are looking for
+        address owner;
+        //@dev the number of vaults a user has, starting at index 0 and incrementing. vaultIds sorted chronologically.
+        uint256 vaultIds;
+    }
+
+    /**
+     * @dev Vault is a struct of 6 arrays that describe a position a user has. A user can have multiple vaults.
+     */
+    struct Vault {
+        /// @dev the addresses of oTokens a user has shorted (i.e. written) against this vault, including those in spreads/combos
+        address[] shortOtokens;
+        /**
+         * @dev the addresses of oTokens a user has gone long (i.e. bought) in this vault, including those in spreads/combos.
+         * Note that a user can be long oTokens without opening a vault (e.g. by buying on a DEX).
+         * Generally, long oTokens will be 'deposited' in vaults to act as collateral in order to write oTokens against (i.e. in spreads/combos).
+         */
+        address[] longOtokens;
+        /// @dev the addresses of other tokens a user has deposited as collateral in this vault
+        address[] collateralAssets;
+        /// @dev describes the quantity of options shorted for each oToken address in shortOtokens
+        uint256[] shortAmounts;
+        /// @dev describes the quantity of options longed in the vault for each oToken address in longOtokens
+        uint256[] longAmounts;
+        /// @dev describes the quantity of tokens deposited as collateral in the vault for each token address in collateralAssets
+        uint256[] collateralAmounts;
+    }
+
+    /**
+     * @dev Increment vault numbers in account
+     * @param _account The account relating to the user for whom the protocol is opening the new vault.
+     */
+    function openNewVault(Account storage _account) internal {
+        _account.vaultIds.add(1);
+    }
+
+    /**
+     * @dev increase the short oToken balance in a vault when a new oToken is minted.
+     * @param _vault The vault the protocol is adding the short position to
+     * @param _shortOtoken The address of the _shortOtoken the protocol is shorting from the user's vault
+     * @param _amount The additional number of _shortOtoken the protocol is shorting from the user's vault
+     * @param _index The index of _shortOtoken in the user's vault.shortOtokens array
+     */
+    function _addShort(
+        Vault storage _vault,
+        address _shortOtoken,
+        uint256 _amount,
+        uint256 _index
+    ) internal {
+        //Check the adding token is the same as vault.shortOtokens[idx] if it exists. (TODO: isn't that the same as the below?)
+        //Reverts if vault.shortOtokens[index] != shortOtoken && vault.shortOtoken[index] !== address(0)
+        require((_vault.shortOtokens[_index] == _shortOtoken) && (_vault.shortOtokens[_index] != address(0)));
```

Comment:
this require statement will not allow anyone to add short option for the first time, as it will check `_vault.shortOtokens[_index] == _shortOtoken`. thoughts @antoncoding @aparnakr ?

it can be changed to 
```
if(_vault.shortOtokens[_index] != address(0)) {
  require(_vault.shortOtokens[_index] == _shortOtoken))
}
```